File: scala-master/scala-master/akka-quickstart-java/src/main/java/com/lightbend/akka/sample/AkkaQuickstart.java
package com.lightbend.akka.sample;

import java.io.IOException;

import com.lightbend.akka.sample.Greeter.Greet;
import com.lightbend.akka.sample.Greeter.WhoToGreet;

import akka.actor.ActorRef;
import akka.actor.ActorSystem;

public class AkkaQuickstart {
  public static void main(String[] args) {
    final ActorSystem system = ActorSystem.create("helloakka");
    try {
      //#create-actors
      final ActorRef printerActor = 
        system.actorOf(Printer.props(), "printerActor");
      final ActorRef howdyGreeter = 
        system.actorOf(Greeter.props("Actor1 dice : ", printerActor), "Actor1");
      final ActorRef helloGreeter = 
        system.actorOf(Greeter.props("Actor2 dice", printerActor), "Actor2");
      final ActorRef goodDayGreeter = 
        system.actorOf(Greeter.props("Actor3 dice", printerActor), "Actor3");
      //#create-actors

      //#main-send-messages
      howdyGreeter.tell(new WhoToGreet("Saludo 1"), ActorRef.noSender());
      howdyGreeter.tell(new Greet(), ActorRef.noSender());

      howdyGreeter.tell(new WhoToGreet("Saludo 2"), ActorRef.noSender());
      howdyGreeter.tell(new Greet(), ActorRef.noSender());

      helloGreeter.tell(new WhoToGreet("Saludo 1"), ActorRef.noSender());
      helloGreeter.tell(new Greet(), ActorRef.noSender());

      goodDayGreeter.tell(new WhoToGreet("Saludo 1"), ActorRef.noSender());
      goodDayGreeter.tell(new Greet(), ActorRef.noSender());
      //#main-send-messages

      System.out.println(">>> Press ENTER to exit <<<");
      System.in.read();
    } catch (IOException ioe) {
    } finally {
      system.terminate();
    }
  }
}

File: scala-master/scala-master/akka-quickstart-scala/src/main/scala/com/example/AkkaQuickstart.scala
//#full-example
package com.example

import akka.actor.{ Actor, ActorLogging, ActorRef, ActorSystem, Props }


//#greeter-companion
//#greeter-messages
object MessageController {
  //#greeter-messages
  def props(message: String, printerActor: ActorRef): Props ={
    Props(new MessageController(message, printerActor))
    //#greeter-messages

  }
  final case class ProcessMsg(who: String)
  case object PrintMsg

}
//#greeter-messages
//#greeter-companion

//#greeter-actor
class MessageController(message: String, printerActor: ActorRef) extends Actor {
  import MessageController._
  import Printer._

  var msgToPrint = ""

  def receive = {
    case ProcessMsg(who) =>
      msgToPrint = message + ", " + who
    case PrintMsg           =>
      //#greeter-send-message
      printerActor ! Printing(msgToPrint)
      //#greeter-send-message
  }
}
//#greeter-actor

//#printer-companion
//#printer-messages
object Printer {
  //#printer-messages
  def props: Props = Props[Printer]
  //#printer-messages
  final case class Printing(msgToPrint: String)
}
//#printer-messages
//#printer-companion

//#printer-actor
class Printer extends Actor with ActorLogging {
  import Printer._

  def receive = {
    case Printing(msgToPrint) =>
      log.info("Greeting received (from " + sender() + "): " + msgToPrint)
  }
}
//#printer-actor

//#main-class
object AkkaQuickstart extends App {
  import MessageController._

  // Create the 'helloAkka' actor system
  val system: ActorSystem = ActorSystem("helloAkka")

  //#create-actors
  // Create the printer actor
  val printerActor: ActorRef = system.actorOf(Printer.props, "printerActor")

  // Create the 'greeter' actors
  val actor1: ActorRef =
    system.actorOf(MessageController.props("Actor1 dice:", printerActor), "actor1")
  val actor2: ActorRef =
    system.actorOf(MessageController.props("Actor2 dice:", printerActor), "actor2")
  val Actor3: ActorRef =
    system.actorOf(MessageController.props("Actor3 dice:", printerActor), "Actor3")
  //#create-actors

  //#main-send-messages
  actor1 ! ProcessMsg("Mi primer saludo")
  actor1 ! PrintMsg

  actor1 ! ProcessMsg("Mi segundo saludo")
  actor1 ! PrintMsg

  actor2 ! ProcessMsg("Mi primer saludo")
  actor2 ! PrintMsg

  Actor3 ! ProcessMsg("Mi primer saludo")
  Actor3 ! PrintMsg
  //#main-send-messages
}
//#main-class
//#full-example
